### PR TITLE
switch build from grunt to gulp

### PR DESCRIPTION
**Pull request by shuwen**
@dlasky @jcmoore

### CODE REVIEW

**Comment by shuwen** [comment]
about to push some WIP stuff for IIFE wrapper—closing this PR until that's done

**Comment by dlasky** [comment]
it would be good to fix the JS/CSS inlining as discussed
definitely need to resolve the clean issue where dist is getting nuked on non-dist local builds

File: Gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+
+gulp.task('copy', function() {
+	return gulp.src([SRC + '**/*.+(html|js|woff)', '!' + SRC +'**/example.html'])
+		.pipe(changed(BUILD))
+		.pipe(debug())
+		.pipe(gulp.dest(BUILD));
+});
+
+gulp.task('sass', function() {
+	var wrapper = fs.readFileSync(TEMPLATES + "style_module_template.html");
```
Comment (anthonykoerber) [inline Gulpfile.js]: @shuwen looks like some strange mangling happening with the style module `<!--.mm-icon *.mm-icon @license.mm-icon *.mm-icon Copyright.mm-icon (c).mm-icon 2015.mm-icon MediaMath.mm-icon Inc..mm-icon All.mm-icon rights.mm-icon reserved..mm-icon *.mm-icon This.mm-icon code.mm-icon may.mm-icon only.mm-icon be.mm-icon used.mm-icon under.mm-icon the.mm-icon BSD.mm-icon style.mm-icon license.mm-icon found.mm-icon at.mm-icon http.mm-icon://mediamath.github.io/strand/LICENSE.txt --.mm-icon> <dom-module.mm-icon id="-style".mm-icon> <template.mm-icon> <style.mm-icon>` etc. Noticed this in docs.

File: Gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+			spare: true
+		}))
+		.pipe(inlinemin())
+		.pipe(header('<!--\n' + fs.readFileSync('BANNER.txt').toString('utf8') + ' -->'))
+		.pipe(changed(DIST))
+		.pipe(gulp.dest(DIST));
+
+	var lib = gulp.src(BUILD + 'strand.html')
+		.pipe(vulcanize({
+			inlineScripts:true,
```
Comment (dlasky) [inline Gulpfile.js]: so the module excludes should actually exclude the mm-components as well, we really only want to inline the styles + js into the markup, but leave all the dependencies as external.

**Comment by dlasky** [comment]
@shuwen if you want to review my changes otherwise i'm pretty happy with merging it

**Comment by shuwen** [comment]
lgtm aside from the 1 note on package.json

**Comment by dlasky** [comment]
the output from dist looks solid with the exception of the now baked styles.html are probably unneeded in the copy. let me know if you want me to tackle this

**Comment by shuwen** [comment]
closing this in favor of #61

**Pull request closed without merging**